### ISSUE TITLE
Definition of path-ref

### ISSUE THREAD

**Issue by italobusi** [tsaad-dev/te#209]
I am not sure the path-ref type is working as expected:
 
```
  typedef path-ref {
    type union {
      type leafref {
        path "/te:te/te:tunnels/te:tunnel/"
           + "te:primary-paths/te:primary-path/te:name";
      }
      type leafref {
        path "/te:te/te:tunnels/te:tunnel/"
           + "te:secondary-paths/te:secondary-path/te:name";
      }
    }
    description
      "This type is used by data models that need to reference
       configured primary or secondary path of a TE tunnel.";
  }
``` 

At least it should use relative paths rather than absolute paths to identify a path within the tunnel the external command has been applied to
                                  
Moreover, how is the union disambiguated? Maybe an additional attribute should be defined

See comment 5) in https://mailarchive.ietf.org/arch/msg/teas/26vHESr_cYIsUes6Fuvw3YAOtjk/

**Comment by italobusi** [comment]
**2022-10-07 TE Call**

Agreed that the XPath needs to be fixed with a relative path. In addition there are two options to solve the union ambiguity issues:
1.	Add one leaf to disambiguate between primary, primary-reverse, secondary or secondary-reverse path-ref
2.	Use a choice between primary-path-ref, primary-reverse-path-ref, secondary-path-ref and secondary-reverse-path-ref
 
@tsaad-dev : What is your preference?

**Comment by italobusi** [comment]
> [TS]: I changed this to string to identify the name of the path that external command refers to. I removed the path-ref union altogether.

I guess we are assuming that the path names are unique within the tunnel. Is it correct? The YANG model does not ensure this condition so I think we need to make it explicit within the draft.

**Comment by tsaad-dev** [comment]
1) path-ref was removed already and path-name was added. 
2) Need to add path-type into TE-TYPEs ID.

**Comment by italobusi** [comment]
> 2\. Need to add path-type into TE-TYPEs ID.

See PR #205

**Comment by tsaad-dev** [comment]
Addressed in 902c1ad
